Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -56,25 +65,167 @@ namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create(Document* document)
+class CookieStore::MainThreadBridge : public ThreadSafeRefCounted<MainThreadBridge, WTF::DestructionThread::Main> {
+public:
+    static Ref<MainThreadBridge> create(CookieStore& cookieStore, ScriptExecutionContextIdentifier contextIdentifier)
+    {
+        return adoptRef(*new MainThreadBridge(cookieStore, contextIdentifier));
+    }
+
+    void get(CookieStoreGetOptions&&, Function<void(CookieStore&, ExceptionOr<Vector<Cookie>>&&)>&&);
+    void getAll(CookieStoreGetOptions&&, URL&&, Function<void(CookieStore&, ExceptionOr<Vector<Cookie>>&&)>&&);
+    void set(CookieInit&& options, Cookie&&, Function<void(CookieStore&, std::optional<Exception>&&)>&&);
+
+private:
+    MainThreadBridge(CookieStore&, ScriptExecutionContextIdentifier);
+
+    void ensureOnMainThread(Function<void(ScriptExecutionContext&)>&&);
+    void ensureOnContextThread(Function<void(CookieStore&)>&&);
+
+    WeakPtr<CookieStore, WeakPtrImplWithEventTargetData> m_cookieStore;
+    ScriptExecutionContextIdentifier m_contextIdentifier;
+};
+
+CookieStore::MainThreadBridge::MainThreadBridge(CookieStore& cookieStore, ScriptExecutionContextIdentifier contextIdentifier)
+    : m_cookieStore(cookieStore)
+    , m_contextIdentifier(contextIdentifier)
```

Comment:
Note that we could get the context identifier from the `cookieStore` directly, we don't need 2 constructor parameters here.